File: 力扣/Week3/src/Wednesday/every.java
package Wednesday;


class TreeNode{
	int val;
	TreeNode right;
    TreeNode left;
    TreeNode(){}
     TreeNode(int val) { this.val = val; }
     TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
     }
}

public class every {
    public TreeNode insertIntoBST(TreeNode root, int val) {
    	TreeNode head=root;
    	TreeNode pre=null;
        if(root == null)
        	return new TreeNode(val);
        while(root != null) {
        	pre=root;
        	if(root.val < val)
        		root=root.right;
        	else if(root.val > val)
        		root=root.left;	
        }
        if(pre.val < val)
        	pre.right=new TreeNode(val);
        else
        	pre.left=new TreeNode(val);
        
        return head;
    }
}

File: 力扣_C++/第六周/Monday/every.cpp
#include<iostream>
#include<math.h>
using namespace std;

struct TreeNode{
	int val;
	TreeNode *left;
	TreeNode *right;
	TreeNode(int x) :val(x), left(NULL), right(NULL){}
};

class every{
public:
	int min = 10000;
	TreeNode *prev = NULL;
	int getMinimumDifference(TreeNode* root) {
		TreeNode* prev = NULL;
		dfs(root);
		return min;
	}

	void dfs(TreeNode* cur){
		if (cur == NULL)
			return;
		dfs(cur->left);
		if (prev){
			int tmp =abs(cur->val - prev->val);
			min = min > tmp ? tmp : min;
		}
		prev = cur;
		dfs(cur->right);
	}
};

int main(){
	TreeNode *h1 = new TreeNode(1);
	TreeNode *h2 = new TreeNode(3);
	TreeNode *h3 = new TreeNode(2);
	h1->right = h2;
	h2->left = h3;
	every t;
	t.getMinimumDifference(h1);
}